lnbits/extensions/boltz/tasks.py

```diff
@@ -1,154 +1,164 @@
+        logger.error(
+            f"Boltz - startup swap check, database is not created yet, do nothing"
+        )
         return
 
     if len(swaps) > 0:
         logger.debug(f"Boltz - {len(swaps)} pending swaps")
         for swap in swaps:
             try:
-                swap_status = get_swap_status(swap)
-                # should only happen while development when regtest is reset
-                if swap_status.exists is False:
-                    logger.debug(f"Boltz - swap: {swap.boltz_id} does not exist.")
-                    await update_swap_status(swap.id, "failed")
-                    continue
-
                 payment_status = await check_transaction_status(
                     swap.wallet, swap.payment_hash
                 )
-
                 if payment_status.paid:
                     logger.debug(
                         f"Boltz - swap: {swap.boltz_id} got paid while offline."
                     )
                     await update_swap_status(swap.id, "complete")
                 else:
-                    if swap_status.hit_timeout:
-                        if not swap_status.has_lockup:
-                            logger.debug(
-                                f"Boltz - swap: {swap.id} hit timeout, but no lockup tx..."
-                            )
+                    try:
+                        _ = client.swap_status(swap.id)
+                    except:
+                        txs = client.mempool.get_txs_from_address(swap.address)
+                        if len(txs) == 0:
                             await update_swap_status(swap.id, "timeout")
                         else:
-                            logger.debug(f"Boltz - refunding swap: {swap.id}...")
-                            await create_refund_tx(swap)
+                            await client.refund_swap(
+                                privkey_wif=swap.refund_privkey,
+                                lockup_address=swap.address,
+                                receive_address=swap.refund_address,
+                                redeem_script_hex=swap.redeem_script,
+                                timeout_block_height=swap.timeout_block_height,
+                            )
                             await update_swap_status(swap.id, "refunded")
-
+            except BoltzNotFoundException as exc:
+                logger.debug(f"Boltz - swap: {swap.boltz_id} does not exist.")
+                await update_swap_status(swap.id, "failed")
+            except MempoolBlockHeightException as exc:
+                logger.debug(
+                    f"Boltz - tried to refund swap: {swap.id}, but has not reached the timeout."
+                )
             except Exception as exc:
-                logger.error(f"Boltz - swap: {swap.id} - {str(exc)}")
+                logger.error(
+                    f"Boltz - unhandled exception, swap: {swap.id} - {str(exc)}"
+                )
 
     if len(reverse_swaps) > 0:
```

Comment:
suggestion: refactor this block into a separate function as well, like the block above

Reply:
4399b4e2